### ISSUE TITLE
Video Prediction in Medical Images

### ISSUE THREAD

**Issue by BraveDistribution** [chengtan9907/OpenSTL#23]
Thanks for the great framework.

I would like to utilize this to forecast the motion of the liver in 2D mri slices. The forecasting seems is training the loss approaching 0.002 on average in val set.

However, the images are really noisy and I don't understand it.

![image](https://github.com/chengtan9907/OpenSTL/assets/36101458/bb1c25f3-959a-488e-8041-fc6bd342c9f3)

My code is as follows: 
`from typing import Any, Optional
import torch
import pytorch_lightning as pl
from pytorch_lightning.utilities.types import STEP_OUTPUT
from openstl.models.simvp_model import SimVP_Model
from torch.nn.functional import mse_loss

class VideoTransformer(pl.LightningModule):
    def __init__(self, ) -> None:
        super().__init__()
        self.model = SimVP_Model((10, 1, 32, 64))
        
        
    def forward(self, x: Any) -> Any:
        y_hat = self.model(x)
        return y_hat
        
    def training_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = mse_loss(y_hat, y)
        self.log('train_loss', loss)
        return loss
    
    def validation_step(self, batch, batch_dix):
        x, y = batch
        y_hat = self(x)
        loss = mse_loss(y_hat, y)
        self.log('val_loss', loss)
        return loss
    
    def configure_optimizers(self) -> Any:
        return torch.optim.Adam(self.parameters(), lr=1e-4)`

**Comment by Lupin1998** [comment]
Hello @BraveDistribution, sorry for the late reply. I have three suggestions for your problem. Firstly, despite the low validation error, you can try to use a larger learning rate (e.g., `1e-3`) to train `SimVP+gSTA`. To my knowledge, SimVP prefers the learning rate around `1e-3` with OneCycle or Consine Annealing scheduler. Secondly, could you please implement the dataloader of your medical image dataset in OpenSTL (e.g., according to [dataloader_human.py](https://github.com/chengtan9907/OpenSTL/blob/master/openstl/datasets/dataloader_human.py)) and perform the experiment again? Maybe I can help you to find out the bugs or issues of your problem when the experiment is conducted fully based on OpenSTL. Thirdly, you might check the visualization code to ensure the ground truth and the prediction frames are using the same normalization strategies and are in the same scale (maybe [vis_video.py](https://github.com/chengtan9907/OpenSTL/tree/master/tools/visualizations/vis_video.py) can help).

**Comment by BraveDistribution** [comment]
Hello @Lupin1998,

I will try to change the learning rate. I probably won't be able to create the dataloader, since the data is private and I can't share it.  I will also check the visualization code you provided and I will keep you in the loop.

**Comment by chengtan9907** [comment]
I will close this issue if there is no more question. Please feel free to open a new issue when you have new questions.
